vizro-core/src/vizro/managers/_model_manager.py

```diff
@@ -46,12 +47,60 @@ def __iter__(self) -> Generator[ModelID, None, None]:
         """
         yield from self.__models
 
+    # TODO: Consider adding an option to iterate only through specific page - "in_page_with_id=None"
     def _items_with_type(self, model_type: Type[Model]) -> Generator[Tuple[ModelID, Model], None, None]:
         """Iterates through all models of type `model_type` (including subclasses)."""
         for model_id in self:
             if isinstance(self[model_id], model_type):
                 yield model_id, cast(Model, self[model_id])
 
+    # TODO: Consider moving this one to the _callback_mapping_utils.py since it's only used there
+    def _get_action_trigger(self, action_id: ModelID) -> VizroBaseModel:  # type: ignore[return]
+        from vizro.models._action._actions_chain import ActionsChain
+
+        for _, actions_chain in model_manager._items_with_type(ActionsChain):
+            if action_id in [action.id for action in actions_chain.actions]:
+                return self[ModelID(str(actions_chain.trigger.component_id))]
+
+    # TODO: Consider returning with yield
+    def _get_model_children(self, model_id: ModelID) -> List[ModelID]:
+        """Gets all child components of the model with `model_id'."""
+        model_children = []
+
+        def __get_model_children_helper(model: VizroBaseModel) -> None:
+            model_children.append(ModelID(str(model.id)))
+            if hasattr(model, "components"):
+                for sub_model in model.components:
+                    __get_model_children_helper(model=sub_model)
+
+        __get_model_children_helper(model=self[model_id])
+        return model_children
+
+    # TODO: Consider moving this method in the Dashboard model or some other util file
+    def _get_model_page(self, model_id: ModelID) -> Page:  # type: ignore[return]
+        """Gets the id of the page containing the model with "model_id"."""
+        from vizro.models import Page
+
+        for page_id, _ in model_manager._items_with_type(Page):
+            page_model_ids = [page_id, self._get_model_children(model_id=page_id)]
+            page: Page = cast(Page, self[page_id])
+
+            if hasattr(page, "actions"):
+                for actions_chain in page._get_page_actions_chains():
+                    page_model_ids.append(actions_chain.id)
+                    for action in actions_chain.actions:
+                        page_model_ids.append(action.id)
+
+            for control in page.controls:
+                page_model_ids.append(control.id)
+                if hasattr(control, "selector") and control.selector:
+                    page_model_ids.append(control.selector.id)
+
+            # TODO: Add navigation, accordions and other page objects
```

Comment:
I would simplify this to a recursive search using something like:

```py
[a for a in dir(page) if not (a.startswith('__') or a.startswith('_') or callable(getattr(page, a)))]
```

which in the case of page finds `['actions', 'components', 'controls', 'description', 'id', 'layout', 'path', 'title']` which we can the feed back again, and always attach if we find a model

Like that this is automatically extensive and we do not have to add things here when we add a new field e.g.

Reply:
Maybe this approach could be comined also with the children finding

Reply:
That sounds promising. I'll test it 👍.